### ISSUE TITLE
The PingPong benchmark does not work as a federated program

### ISSUE THREAD

**Issue by cmnrd** [icyphy/lingua-franca#515]
I tried to run a federated version of PingPong, but it does not seem to work. I simply changed `main reactor` to `federated reactor` and compiled again. The resulting program starts, but does not seem to make progress. Even when reducing the ping pong messages to a single one, the program does not terminate. I thought at first that this might be an issue with the zero-delay loop, but also when I add after delays to the connections, it does not work.

**Comment by Soroosh129** [comment]
This is most likely due to `fast: true`.

**Comment by cmnrd** [comment]
Oh, ok. Shouldn't there be a warning/error then?

Without `fast`, I get this error (for centralized control and with an after delay of 10ms):
```
Federate 0: Starting timestamp is: 1632150601310139005.
Federate 1: Starting timestamp is: 1632150601310139005.
Federate 1: FATAL ERROR: Received a message at tag (30000000, 0) that has a tag (30000000, 0) that has violated the STP offset. Centralized coordination should not have these types of messages.
Federate 1: WARNING: Memory allocated for tokens has not been freed!
Federate 1: WARNING: Number of unfreed tokens: 1.
Federate 1: ---- Elapsed logical time (in nsec): 30,000,000
Federate 1: ---- Elapsed physical time (in nsec): 31,146,395
Federate 1 has resigned.
```

**Comment by Soroosh129** [comment]
> Oh, ok. Shouldn't there be a warning/error then?

Adjusted the error reporting in e27ce4164c46d36a1661435540faf5a0172fd3c6.

> Without `fast`, I get this error (for centralized control and with an after delay of 10ms):
> 
> ```
> Federate 0: Starting timestamp is: 1632150601310139005.
> Federate 1: Starting timestamp is: 1632150601310139005.
> Federate 1: FATAL ERROR: Received a message at tag (30000000, 0) that has a tag (30000000, 0) that has violated the STP offset. Centralized coordination should not have these types of messages.
> Federate 1: WARNING: Memory allocated for tokens has not been freed!
> Federate 1: WARNING: Number of unfreed tokens: 1.
> Federate 1: ---- Elapsed logical time (in nsec): 30,000,000
> Federate 1: ---- Elapsed physical time (in nsec): 31,146,395
> Federate 1 has resigned.
> ```

I have seen this in the Autoware port as well. We will have to investigate this.

**Comment by lhstrh** [comment]
@Soroosh129: has an issue been created for this?

**Comment by Soroosh129** [comment]
> @Soroosh129: has an issue been created for this?

No. This has slipped through the bugs that I saw and fixed while developing that project.

**Comment by lhstrh** [comment]
Could you create one? Thanks!
